internal/dataplane/kongstate/util.go

```diff
@@ -141,11 +141,12 @@ func kongPluginFromK8SClusterPlugin(
 		Name:   k8sPlugin.PluginName,
 		Config: config,
 
-		RunOn:     k8sPlugin.RunOn,
-		Ordering:  k8sPlugin.Ordering,
-		Disabled:  k8sPlugin.Disabled,
-		Protocols: protocolsToStrings(k8sPlugin.Protocols),
-		Tags:      util.GenerateTagsForObject(&k8sPlugin),
+		RunOn:        k8sPlugin.RunOn,
+		Ordering:     k8sPlugin.Ordering,
+		InstanceName: k8sPlugin.InstanceName,
```

Comment:
Have we tested the situation where Kong does not support `instance_name` field?

Reply:
It'll be rejected, but only if you set it. [The field is `omitempty`](https://github.com/Kong/go-kong/blob/a49e9161e9667bbcc67eaf0a871c60f9a124cbe9/kong/plugin.go#L10) and just won't be included if it's the zero value.

We don't usually have any special gates around optional fields with version constraints, since it's simpler to just say "don't try to set it if you can't". `Ordering` is the same way, where it's only allowed if you have Enterprise 3.0+.

For comparison, I ran 3.0.2, created a plugin without `instance_name`, and then added it after: [instance_unsupported.txt](https://github.com/Kong/kubernetes-ingress-controller/files/11749742/instance_unsupported.txt)